R/summarize_patients_exposure_in_cols.R

```diff
@@ -136,28 +179,87 @@ summarize_patients_exposure_in_cols <- function(lyt, # nolintr
                                                 .labels = c(n_patients = "Patients", sum_exposure = "Person time"),
                                                 .indent_mods = NULL,
                                                 col_split = TRUE) {
-  afun_list <- Map(
-    function(stat) {
-      make_afun(
-        s_count_patients_sum_exposure,
-        .stats = stat,
-        .formats = ifelse(stat == "n_patients", "xx (xx.x%)", "xx")
-      )
-    },
-    stat = .stats
-  )
-
   if (col_split) {
     lyt <- split_cols_by_multivar(
       lyt = lyt,
       vars = rep(var, length(.stats)),
-      varlabels = .labels[.stats]
+      varlabels = .labels[.stats],
+      extra_args = list(.stats = .stats)
     )
   }
   summarize_row_groups(
     lyt = lyt,
     var = var,
-    cfun = afun_list,
+    cfun = a_count_patients_sum_exposure,
     extra_args = list(...)
   )
 }
+
+#' @describeIn summarize_patients_exposure_in_cols Layout-creating function which can take statistics
+#'   function arguments and additional format arguments. This function is a wrapper for
+#'   [rtables::split_cols_by_multivar()] and [rtables::analyze()].
+#'
+#' @inheritParams argument_convention
+#' @param col_split (`flag`)\cr whether the columns should be split. Set to `FALSE` when the required
+#'   column split has been done already earlier in the layout pipe.
+#'
+#' @return
+#' * `analyze_patients_exposure_in_cols()` returns a layout object suitable for passing to further
+#'   layouting functions, or to [rtables::build_table()]. Adding this function to an `rtable` layout will
+#'   add formatted data rows, with the statistics from `s_count_patients_sum_exposure()` arranged in
+#'   columns, to the table layout.
+#'
+#' @note As opposed to [summarize_patients_exposure_in_cols()] which generates content rows, this function
+#'   generates data rows which will _not_ be repeated on multiple pages when pagination is used.
+#'
+#' @examples
+#' lyt3 <- basic_table() %>%
+#'   split_cols_by("ARMCD", split_fun = add_overall_level("Total", first = FALSE)) %>%
+#'   summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE) %>%
+#'   analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE)
+#' result3 <- build_table(lyt3, df = df, alt_counts_df = adsl)
+#' result3
+#'
+#' lyt4 <- basic_table() %>%
+#'   split_cols_by("ARMCD", split_fun = add_overall_level("Total", first = FALSE)) %>%
+#'   summarize_patients_exposure_in_cols(
+#'     var = "AVAL", col_split = TRUE,
+#'     .stats = "n_patients", custom_label = "some custom label"
+#'   ) %>%
+#'   analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE)
+#' result4 <- build_table(lyt4, df = df, alt_counts_df = adsl)
+#' result4
+#'
+#' lyt5 <- basic_table() %>%
+#'   summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE)
```

Comment:
Maybe here you meant analyze?

Reply:
do you meant @Melkiades 
```suggestion
#'   summarize_patients_exposure_in_cols(var = "AVAL", col_split = TRUE) %>%
#'   analyze_patients_exposure_in_cols(var = "SEX", col_split = FALSE)
```

Reply:
or did you mean? @Melkiades 

```
> lyt5 <- basic_table() %>%
+     analyze_patients_exposure_in_cols(var = "SEX", col_split = TRUE)
> result5 <- build_table(lyt5, df = df, alt_counts_df = adsl)
> result5

         Patients    Person time
————————————————————————————————
Female   6 (50.0%)       46     
Male     6 (50.0%)       68 
```

Reply:
the naming, probably can be more consistent with other functions. In other functions, `var` is the variable being analyzed, but here it is rather a "split" variable?

Reply:
hey @clarkliming , but for this function it was explicitly looking for exposure variable, which is defined by ex_var, var is used only for splitting rows